pkg/upstream/http.go

```diff
@@ -100,17 +101,37 @@ func newReverseProxy(target *url.URL, upstream options.Upstream, errorHandler Pr
 	if upstream.FlushInterval != nil {
 		proxy.FlushInterval = *upstream.FlushInterval
 	} else {
-		proxy.FlushInterval = 1 * time.Second
+		proxy.FlushInterval = options.DefaultFlushInterval
 	}
 
+	var finalTimeout time.Duration
+	if upstream.Timeout != nil {
+		finalTimeout = *upstream.Timeout
+	} else {
+		finalTimeout = options.DefaultTimeout
+	}
+
+	var finalKeepAlive time.Duration
+	if upstream.Timeout != nil {
+		finalKeepAlive = *upstream.KeepAlive
+	} else {
+		finalKeepAlive = options.DefaultKeepAlive
+	}
+
+	defaultTransportClone := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Style NIT: I guess we don't really care that our initialization method is a clone of the default transport to be information carried on with the naming of the variable.  I just care its the transport we're using.

I would name the variable something simpler (ie `transport` since it is ours moving forward) to cut down verbose-ness